plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/models/OpenshiftUIModel.java

```diff
@@ -0,0 +1,189 @@
+			}
+
+			@Override
+			public void connectionAdded(IConnection connection) {
+				if (!(connection instanceof IOpenShiftConnection)) {
+					return;
+				}
+				synchronized (connections) {
+					if (connections.containsKey(connection)) {
+						return;
+					}
+					connections.put((IOpenShiftConnection) connection,
+							new ConnectionWrapper(OpenshiftUIModel.this, (IOpenShiftConnection) connection));
+				}
+				fireChanged();
+			}
+		};
+		Collection<IConnection> allConnections = registry.getAll();
+		synchronized (connections) {
+			for (IConnection connection : allConnections) {
+				if (connection instanceof IOpenShiftConnection) {
+					connections.put((IOpenShiftConnection) connection,
+							new ConnectionWrapper(OpenshiftUIModel.this, (IOpenShiftConnection) connection));
+				}
+			}
+		}
+		registry.addListener(listener);
+	}
+	public static boolean isOlder(IResource oldResource, IResource newResource) {
+		try {
+			int oldVersion = Integer.valueOf(oldResource.getResourceVersion());
+			int newVersion = Integer.valueOf(newResource.getResourceVersion());
+			return oldVersion < newVersion;
+		} catch (NumberFormatException e) {
+			// treat this as an update
+			return true;
+		}
+	}
+
+	@Override
+	public IOpenshiftUIElement<?> getParent() {
+		return null;
+	}
+
+	@Override
+	public void fireChanged(IOpenshiftUIElement<?> source) {
+		if (Display.getCurrent() != null) {
+			dispatchChange(source);
+		} else {
+			Display.getDefault().asyncExec(new Runnable() {
+
+				@Override
+				public void run() {
+					dispatchChange(source);
+				}
+			});
+		}
+	}
+
+	protected void dispatchChange(IOpenshiftUIElement<?> source) {
+		Collection<IElementListener> copy = new ArrayList<>();
+		synchronized (listeners) {
+			copy.addAll(listeners);
+		}
+		copy.forEach(l -> l.elementChanged(source));
+	}
+
+	public void addListener(IElementListener l) {
+		synchronized (listeners) {
+			listeners.add(l);
+		}
+	}
+
+	public void removeListener(IElementListener l) {
+		synchronized (listeners) {
+			int lastIndex = listeners.lastIndexOf(l);
+			if (lastIndex >= 0) {
+				listeners.remove(lastIndex);
+			}
+		}
+	}
+
+	@Override
+	public OpenshiftUIModel getRoot() {
+		return this;
+	}
+
+	public Collection<ConnectionWrapper> getConnections() {
+		synchronized (connections) {
+			return new ArrayList<ConnectionWrapper>(connections.values());
+		}
+	}
+
+	public void dispose() {
+		ConnectionsRegistrySingleton.getInstance().removeListener(listener);
+	}
+
+	@Override
+	public void refresh() {
+			Map<IOpenShiftConnection, ConnectionWrapper> updated = new HashMap<>();
+			boolean changed = false;
+			synchronized (connections) {
+				HashMap<IOpenShiftConnection, ConnectionWrapper> oldWrappers = new HashMap<>(connections);
+				connections.clear();
+				for (IConnection r : ConnectionsRegistrySingleton.getInstance().getAll()) {
+					if (!(r instanceof IOpenShiftConnection)) {
+						return;

```

Comment:
Probably 'continue', but the check will not be needed, when registry.getAll(Connection.class) is used.